### PR TITLE
Add support for snake case in config.yml. Add support for default_version field

### PR DESCRIPTION
**Pull request by amy**

### CODE REVIEW

File: model/template.go [hunk 1]
```diff
@@ -16,11 +16,11 @@ type Template struct {
 	Name           string `json:"name"`
 	IsSystem       string `json:"isSystem"`
 	Description    string `json:"description"`
-	DefaultVersion string `json:"defaultVersion" yaml:"version"`
+	DefaultVersion string `json:"defaultVersion" yaml:"default_version"`
```
Comment (joshwget) [inline model/template.go]: @deniseschannon Should we change this to `default_version`? Or just leave as `version`?

File: model/template.go [hunk 2]
```diff
@@ -16,11 +16,11 @@ type Template struct {
...
 	Path           string `json:"path"`
 	Maintainer     string `json:"maintainer"`
 	License        string `json:"license"`
-	ProjectURL     string `json:"projectURL" yaml:"projectURL"`
+	ProjectURL     string `json:"projectURL" yaml:"project_URL"`
```
Comment (joshwget) [inline model/template.go]: `project_url`

File: manager/traverse.go [hunk 3]
```diff
@@ -263,10 +264,27 @@ func handleFile(templateIndex map[string]*model.Template, fullPath, relativePath
 		if err != nil {
 			return err
 		}
+
```
Comment (joshwget) [inline manager/traverse.go]: Can we move this block to `parse.CatalogInfoFromTemplateVersion`?

Comment (joshwget) [inline manager/traverse.go]: Wait no, wouldn't want it in that function. Something like `parse.TemplateInfo` might be good though.

File: model/template.go (hunk 1 above)
Comment (joshwget) [inline model/template.go]: I guess we decided yes to this at some point since it's in rancher/rancher#8319.

File: manager/traverse.go [hunk 4]
```diff
@@ -252,6 +253,32 @@ func traverseGitFiles(repoPath string) ([]model.Template, []error, error) {
 	return templates, errors, nil
 }
 
+func templateInfo(contents []byte) (model.Template, error) {
```
Comment (joshwget) [inline manager/traverse.go]: Can you move this to the parse package?

**Comment by joshwget** [comment]
LGTM!

**Comment by cjellick** [comment]
LGTM

**Pull request merged**